pkg/ccl/backupccl/split_and_scatter_processor.go

```diff
@@ -95,13 +98,46 @@ func (s dbSplitAndScatterer) split(
 		newSplitKey = splitAt
 	}
 	log.VEventf(ctx, 1, "presplitting new key %+v", newSplitKey)
-	if err := s.db.AdminSplit(ctx, newSplitKey, expirationTime); err != nil {
+	if err := s.adminSplitWithRetry(ctx, newSplitKey, expirationTime); err != nil {
 		return errors.Wrapf(err, "splitting key %s", newSplitKey)
 	}
 
 	return nil
 }
 
+func (s dbSplitAndScatterer) adminSplitWithRetry(
+	ctx context.Context, newSplitKey roachpb.Key, expirationTime hlc.Timestamp,
+) error {
+
+	retryOpts := retry.Options{
+		MaxBackoff: 1 * time.Second,
+		MaxRetries: 5,
+	}
+
+	var err error
+	for r := retry.StartWithCtx(ctx, retryOpts); r.Next(); {
+
+		if err == nil {
+			break
+		}
+		err := s.db.AdminSplit(ctx, newSplitKey, expirationTime)
+
+		if !(kvserver.IsRetriableReplicationChangeError(err)) {
```

Comment:
We might need to add `IsLeaseTransferRejectedBecauseTargetMayNeedSnapshotError()` here or inside some lower level e.g. `IsRetriableReplicationChangeError()` as this error is what #84891 is failing with.